p2p/ping.go

```diff
@@ -79,50 +79,53 @@ func pingPeer(ctx context.Context, svc *ping.PingService, p peer.ID, callback fu
 	}
 }
 
-// pingPeerOnce starts a long lived ping connection with the peer and returns on first error.
+// pingPeerOnce starts a long-lived ping connection with the peer and returns on first error.
 func pingPeerOnce(ctx context.Context, svc *ping.PingService, p peer.ID,
 	logFunc func(context.Context, ping.Result), callback func(peer.ID, host.Host),
 ) {
-	ctx, cancel := context.WithCancel(ctx)
-	defer cancel()
-
 	// newPingChan creates a new stream and returns the ping result channel to listen for results
 	// and a close function to close the stream.
 	// Note: The only way to close a stream opened by ping service is to cancel the context.
-	newPingChan := func() (<-chan ping.Result, func()) {
-		pingCtx, pingCancel := context.WithCancel(ctx)
+	newPingChan := func(parentCtx context.Context) (<-chan ping.Result, func()) {
```

Comment:
nit: can simply name `ctx`
```suggestion
	newPingChan := func(ctx context.Context) (<-chan ping.Result, func()) {
```